compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1413,6 +1424,23 @@ object Parsers {
       }
     }
 
+    /** Replaces kind-projector's `*` in a list of types arguments with synthetic names,
+     *  returning the new argument list and the synthetic type definitions.
+     */
+    private def replaceKindProjectorPlaceholders(params: List[Tree]): (List[Tree], List[TypeDef]) = {
+      val tparams = new ListBuffer[TypeDef]
+
+      val newParams = params.map {
```

Comment:
To reuse the same list if possible:
```suggestion
      val newParams = params.mapConserve {
```